cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -82,7 +82,43 @@ def add_sample_lane_sequencing_metrics_for_flow_cell(self, flow_cell_name: str):
         LOG.info(f"Added sample lane sequencing metrics to status database for: {flow_cell_name}")
 
     def finish_flow_cell_temp(self, flow_cell_name: str) -> None:
-        """Finish flow cell."""
+        """
+        1. Validate that the flow cell directory exists.
+        2. Validate that the demultiplexing data transfer is complete.
+        3. Create flow cell.
+        4. Store flow cell in status db.
+        5. Store flow cell data in housekeeper.
+        6. Create sequencing metrics.
+        """
+
+        LOG.info(f"Finish flow cell {flow_cell_name}")
+
+        flow_cell_dir: Path = Path(self.demux_api.out_dir, flow_cell_name)
+        bcl_converter: str = self.get_bcl_converter(flow_cell_name=flow_cell_name)
+
+        # 1. Validate that the flow cell directory exists.
+        if not flow_cell_dir.exists():
+            return
+
+        # 2. Validate that the demultiplexing data transfer is complete.
+        if not Path(flow_cell_dir, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists():
+            return
+
+        # 3. Create flow cell. TODO: Populate all fields here.
+        try:
+            flow_cell: FlowCell = FlowCell(
+                flow_cell_path=flow_cell_dir, bcl_converter=bcl_converter
+            )
+        except FlowCellError:
+            return None
+
+        # 4. Store flow cell in status db.
+        self.status_db.session.add(flow_cell)
+        self.status_db.session.commit()
```

Comment:
I think these are all the steps which are needed indeed. Would be nice to separate them into smaller functions to keep them more testable and simple.

Reply:
Yes, that sounds like a good idea :)

Reply:
I'll keep this outline for now and refactor each step in separate PR:s.